lib/sensors/src/UltraSonicSensor.cpp

```diff
@@ -0,0 +1,33 @@
+#include "UltraSonicSensor.h"
+
+UltraSonicSencor::UltraSonicSensor(PinName echoPin, PinName trigPin) : _echoPin(echoPin), _trigPin(trigPin) {}
+
+uint64_t UltraSonicSensor::pulseIn() {
+  uint64_t _time{};
+  // instantiate PwmIn class
+  PwmIn _pulse(_echopin, NUM_SAMPLES);
```

Comment:
Why construct this object every single time you call get distance? You can construct a pwmIn object once and have it be a private member of the class. Assuming you implement that, this function becomes one line and you might as well just have it in getDistance.